go/adbc/driver/flightsql/flightsql_connection.go

```diff
@@ -57,10 +57,63 @@ type connectionImpl struct {
 	supportInfo support
 }
 
+type flightSqlMetadata struct {
+	internal.DefaultXdbcMetadataBuilder
+}
+
+func (md *flightSqlMetadata) SetXdbcScopeCatalog(b *array.StringBuilder) {
+	f := &flightsql.ColumnMetadata{Data: md.Metadata()}
```

Comment:
rather than doing this on every call here, why not have the `SetMetadata` method create this and make it a member of `flightSqlMetadata` ?